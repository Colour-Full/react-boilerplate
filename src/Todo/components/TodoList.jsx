// @flow
import React, { type Element } from 'react'
import { type TodoTypeFromReducer } from '../api-todo/reducers/todo'
import Todo from './ToDo'
import { ItemGroup } from '@atlaskit/item'
import { Grid, GridColumn } from '@atlaskit/page'

export type TodoListProps = {
  todos: Array<TodoTypeFromReducer>,
  toggleTodo: (number) => any
}

const TodoList = ({ todos, toggleTodo }: TodoListProps): Element<any> => (
  <Grid>
    <GridColumn medium={4}>
      <ItemGroup>
        {todos.map((todo: TodoTypeFromReducer): Element<any> =>
          <Todo
            key={todo.id}
            completed={todo.completed}
            text={todo.text}
            onClick={(): Object => toggleTodo(todo.id)}
          />
        )}
      </ItemGroup>
    </GridColumn>
  </Grid>
)

export default TodoList
