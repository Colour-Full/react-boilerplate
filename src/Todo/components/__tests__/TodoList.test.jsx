// @flow
import React from 'react'
import TodoList from '../TodoList'
import Adapter from 'enzyme-adapter-react-16'
import Enzyme, { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'

Enzyme.configure({ adapter: new Adapter() })

describe('TodoList component', () => {
  it('Should render without throwing an error', () => {
    const todos = [
      { id: 0, text: 'First Thing To Do', completed: false },
      { id: 1, text: 'Second Thing To Do', completed: false },
      { id: 2, text: 'Third Thing To Do', completed: false }
    ]
    const toggleTodo = jest.fn()
    const wrapper = shallow(<TodoList todos={todos} toggleTodo={toggleTodo} />)
    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })
})
