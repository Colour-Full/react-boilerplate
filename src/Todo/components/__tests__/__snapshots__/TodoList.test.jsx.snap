// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoList component Should render without throwing an error 1`] = `
<WithTheme(AkGrid)>
  <WithTheme(AkGridColumn)
    medium={4}
  >
    <ItemGroup
      role="group"
    >
      <Todo
        completed={false}
        key="0"
        onClick={[Function]}
        text="First Thing To Do"
      />
      <Todo
        completed={false}
        key="1"
        onClick={[Function]}
        text="Second Thing To Do"
      />
      <Todo
        completed={false}
        key="2"
        onClick={[Function]}
        text="Third Thing To Do"
      />
    </ItemGroup>
  </WithTheme(AkGridColumn)>
</WithTheme(AkGrid)>
`;
